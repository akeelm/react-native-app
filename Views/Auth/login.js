import React from 'React';
import { Image, StyleSheet, Text, View, Button, TextInput, ScrollView } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import TextInputLogo from '../../Components/Controls/text-input-logo.js';
import PropTypes from 'prop-types';
import MainStyles from '../../Styles/main.js';

export default class Login extends React.Component {
	render() {

		return (
			<View style={MainStyles.defaultbackground}>

				<View style={{ 
						flex: 0.4, 
						flexDirection: 'column',
						alignItems: 'center', 
						justifyContent: 'flex-start', 
						paddingTop: 20
				}}>
					<Image 
						source={require('../../Images/logo.png')} 
						style={MainStyles.logoImage}
					/>
				</View>

				<KeyboardAwareScrollView enableOnAndroid={true} >
					<View style={{ 
							flex: 1, 
							flexDirection: 'column',
							alignItems: 'stretch', 
							justifyContent: 'flex-start', 
							padding: 40
					}}>

						<TextInputLogo title='Username' logo='user-o' />

						<TextInputLogo title='Password' logo='lock' secure={true} />


						<Text style={{ 
							alignSelf: 'flex-end',
							paddingTop: 10,
							paddingBottom: 20,
							color: '#4e4e4e'
						}}>Forgot Password?</Text>
						<Button 
							title='Login' 
							onPress={() => { }}
							style={{ 
								paddingTop: 10,
								backgroundColor: '#ccf'
							}}
						>
							Login
						</Button>

					</View>
				</KeyboardAwareScrollView>

			</View>
		);
	}
}
