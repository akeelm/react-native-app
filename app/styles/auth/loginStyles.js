import { StyleSheet } from 'react-native';

export default StyleSheet.create({
	logoView: {
        flex: 0.4,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'flex-start',
        paddingTop: 20
	},
	loginFormView: {
        flex: 1,
        flexDirection: 'column',
        alignItems: 'stretch',
        justifyContent: 'flex-start',
        padding: 40
    },
    forgotPassword: {
        alignSelf: 'flex-end',
        paddingTop: 10,
        paddingBottom: 20,
        color: '#4e4e4e'
    },
    loginButton : {
        paddingTop: 10,
        backgroundColor: '#ccf'
    }
});

