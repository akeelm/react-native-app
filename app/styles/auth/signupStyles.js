import { StyleSheet } from 'react-native';

export default StyleSheet.create({
	signupView: {
		flex: 0.2,
		flexDirection: 'column',
		alignItems: 'center',
		justifyContent: 'flex-start', 
	},
	signupLogo: {
		height: 60,
		fontSize: 25,
		paddingTop: 20
	},
	signupForm: {
		flex: 1,
		flexDirection: 'column',
		alignItems: 'stretch',
		justifyContent: 'flex-start',
		padding: 40
	},
	signupButton: {
		paddingTop: 10,
		backgroundColor: '#ccf'
	}
});

