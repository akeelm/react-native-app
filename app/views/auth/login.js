import React from 'React';
import { connect } from 'react-redux';
import { Image, StyleSheet, Text, View, Button, TextInput, ScrollView } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import TextInputLogo from '../../components/controls/text-input-logo';
import PropTypes from 'prop-types';
import MainStyles from '../../styles/mainStyles';
import LoginStyles from '../../styles/auth/loginStyles';

import Icon from 'react-native-vector-icons/FontAwesome';

import { loginUser } from './../../actions';

import {
	_validateEmail,
	_minimumPasswordLength
} from '../../helpers/validation';

class Login extends React.Component {
	constructor(props) {
		super(props);

		this._onChangeUsername = this._onChangeUsername.bind(this);
		this._onChangePassword = this._onChangePassword.bind(this);

		this.state = { formValid: false };
	}

	loginButtonPress() {
		this.props.loginUser({ username: this.state.username, password: this.state.password });
	}

	_onChangeUsername(username) {
		this.setState(Object.assign({}, this.state, { username: username }));
	}

	_onChangePassword(password) {
		this.setState(Object.assign({}, this.state, { password: password, formValid: true }));
	}


	render() {

		return (
			<View style={MainStyles.defaultbackground}>


				<View style={LoginStyles.logoView}>
					<Image 
						source={require('../../images/logo.png')} 
						style={MainStyles.logoImage}
					/>
				</View>

				<KeyboardAwareScrollView enableOnAndroid={true} >
					<View style={LoginStyles.loginFormView}>

						<TextInputLogo title='E-mail' logo='user-o' 
							onChangeMethod={this._onChangeUsername}
							validate={_validateEmail} 
							/>

						<TextInputLogo title='Password' logo='lock' secure={true} 
							onChangeMethod={this._onChangePassword}
							validate={_minimumPasswordLength} 
							/>

						<Text style={LoginStyles.forgotPassword}>Forgot Password?</Text>

						<Button 
							title='Login' 
							onPress={() => { this.loginButtonPress() }}
							style={LoginStyles.loginButton}
							disabled={!this.state.formValid}
						>
							Login
						</Button>

					</View>
				</KeyboardAwareScrollView>

			</View>
		);
	}
}

const mapStateToProps = state => ({ });

export default connect(mapStateToProps, { loginUser } )(Login)