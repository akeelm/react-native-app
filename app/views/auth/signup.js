import React from 'React';
import { 
	StyleSheet, 
	Text, 
	View, 
	Button, 
	TextInput, 
	ScrollView, 
	Platform
} from 'react-native';
import TextInputLogo from '../../components/controls/text-input-logo';
import PropTypes from 'prop-types';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import MainStyles from '../../styles/mainStyles';
import SignupStyles from '../../styles/auth/signupStyles';
import {
	_validateEmail,
	_minimumPasswordLength,
	_passwordsMustMatch
} from '../../helpers/validation';

export default class Signup extends React.Component {
	constructor(props) {
		super(props);

		this._onChangeEmail = this._onChangeEmail.bind(this);
		this._onChangePassword = this._onChangePassword.bind(this);
		this._onChangeConfirmPassword = this._onChangeConfirmPassword.bind(this);
		this.state = { password : '', formValid: false };
	}

	_onChangeEmail(username) {
		this.setState(Object.assign({}, this.state, { username: username }));
	}

	_onChangePassword(password) {
		this.setState(Object.assign({}, this.state, { password: password }));
	}

	_onChangeConfirmPassword(password) {
		this.setState(Object.assign({}, this.state, { confirmPassword: password, formValid: true }));
	}


	render() {

		return (
			<View style={MainStyles.defaultbackground}>

				<View style={SignupStyles.signupView}>
					<Text style={SignupStyles.signupLogo}>
						Sign up
					</Text>
				</View>

				<KeyboardAwareScrollView enableOnAndroid={true} >
					<View style={SignupStyles.signupForm}>

							<TextInputLogo title='E-mail' logo='envelope-o'
								validate={_validateEmail} 
								onChangeMethod={this._onChangeEmail} />

							<TextInputLogo title='Password' logo='lock' secure={true}
								validate={_minimumPasswordLength}
								onChangeMethod={this._onChangePassword} />

							<TextInputLogo title='Confirm password' logo='lock' secure={true} 
								validate={_passwordsMustMatch}
								onChangeMethod={this._onChangeConfirmPassword} 
								checkValue={this.state.password} />

							<Button 
								title='Confirm' 
								onPress={() => { }}
								style={SignupStyles.signupButton}
								disabled={!this.state.formValid}
							></Button>

					</View>
				</KeyboardAwareScrollView>

			</View>
		);
	}
}