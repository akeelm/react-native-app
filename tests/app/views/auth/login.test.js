import React from 'react';
import { Login } from '../../../../app/views/auth/login';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const rendered = renderer.create(<Login />).toJSON();
    expect(rendered).toMatchSnapshot();
});