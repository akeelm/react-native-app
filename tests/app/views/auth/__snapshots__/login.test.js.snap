// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f3f7ff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.4,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "paddingTop": 20,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 75,
          "width": 270,
        }
      }
    />
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={undefined}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "padding": 40,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 15,
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <TextInput
              allowFontScaling={true}
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="E-mail"
              returnKeyType="next"
              secureTextEntry={undefined}
              style={
                Object {
                  "borderBottomWidth": 0,
                  "flex": 1,
                  "height": 40,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 15,
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <TextInput
              allowFontScaling={true}
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Password"
              returnKeyType="next"
              secureTextEntry={true}
              style={
                Object {
                  "borderBottomWidth": 0,
                  "flex": 1,
                  "height": 40,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#4e4e4e",
              "paddingBottom": 20,
              "paddingTop": 10,
            }
          }
        >
          Forgot Password?
        </Text>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
              "disabled",
            ]
          }
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#cdcdcd",
                  },
                ]
              }
            >
              Login
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
